includes/class-wc-payments-account.php

```diff
@@ -110,6 +110,28 @@ public function is_stripe_connected( bool $on_error = false ): bool {
 		}
 	}
 
+	/**
+	 * Checks if the account has been rejected, assumes the value of $on_error on server error.
+	 *
+	 * @param bool $on_error Value to return on server error, defaults to false.
+	 *
+	 * @return bool True if the account is rejected, false otherwise, $on_error on error.
+	 */
+	public function is_account_rejected( bool $on_error = false ): bool {
+		try {
+			$account = $this->get_cached_account_data();
+
+			if ( empty( $account ) ) {
+				// Empty means no account, so not rejected.
+				return false;
+			}
+
+			return strpos( $account['status'], 'rejected' ) === 0;
+		} catch ( Exception $e ) {
```

Comment:
Is there any particular known way that an exception would be thrown?

Reply:
Hm, I see how it could be considered an error if the account doesn't exist but actually it's not obvious to me that it shouldn't be false, since the function name could possibly be interpreted as "is there a connected account that is rejected" – and that's also closer to what I intended with the name of a new function `is_account_onboarded` in https://github.com/Automattic/woocommerce-payments/pull/3767/commits/275c5077308da15e654cdd524beee907575cb4e5 (from a draft PR).

What do you think? If it does return an error, then the function comment should be expanded beyond "on server error".

Reply:
Sounds good 👍